src/main/java/software/amazon/event/kafkaconnector/batch/BestEffortInOrderEventBridgeBatching.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class BestEffortInOrderEventBridgeBatching implements EventBridgeBatchingStrategy {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(BestEffortInOrderEventBridgeBatching.class);
+
+  private final Collector<
+          MappedSinkRecord<PutEventsRequestEntry>,
+          PutEventsRequestEntrySizeCollector,
+          Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>>
+      collector =
+          Collector.of(
+              PutEventsRequestEntrySizeCollector::new,
+              PutEventsRequestEntrySizeCollector::accumulate,
+              (left, right) -> {
+                throw new IllegalStateException();
+              },
+              PutEventsRequestEntrySizeCollector::finish);
+
+  @Override
+  public Stream<List<MappedSinkRecord<PutEventsRequestEntry>>> apply(
+      Stream<MappedSinkRecord<PutEventsRequestEntry>> records) {
+    if (records.isParallel()) {
+      throw new IllegalArgumentException("Stream must not be parallel.");
+    }
+    return records.collect(collector);
+  }
+
+  static class PutEventsRequestEntrySizeCollector {
+
+    private static final int MAX_BATCH_SIZE_BYTES = 256 * 1024;
+
+    private final List<List<MappedSinkRecord<PutEventsRequestEntry>>> batches = new ArrayList<>();
+    private int actualBatchSize = 0;
+    private int index = 0;
+
+    PutEventsRequestEntrySizeCollector() {
+      batches.add(new ArrayList<>());
+    }
+
+    void accumulate(final MappedSinkRecord<PutEventsRequestEntry> item) {
+
+      var itemSize = getSize(item.getValue());
+      if (itemSize > MAX_BATCH_SIZE_BYTES) {
```

Comment:
@agebhar1 Maybe I'm overlooking it but where do we check for the max batch size of 10?

Reply:
Ah, good catch man!

Reply:
It's not yet implemented. Can you please explain the intention for restricting the batch size by count. If there are a lot of small (bytes) items and they are less (equal) than 256kb at all, why should I send them in multiple batches? E.g. 256 items each with 1Kb, 26 batches vs. 1 batch.

Reply:
Because Entries is limited to a maximum of 10 items. See: https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutEvents.html#API_PutEvents_RequestParameters

Reply:
I missed that info 🤣. Thank's @maschnetwork for the information. I will add the link to the implementation documentation.